### ISSUE TITLE
[BUG] Collection serialization 3.2.0rc5

### ISSUE THREAD

**Issue by Nordran** [elsa-workflows/elsa-core#5922]
Hello.
I've encountered a problem with the List type WorkflowStorageDriver variable in version 3.2.0rc5. Its value has been serialized incorrectly.
v3.2.0rc5:
```json
"PersistentVariablesDictionary": {
  "b634f386a7354d7cbe853dd8807cb6cd": {
	  "\\\\$values": {
		  "\\\\$values": {
			  "\\\\$values": {
				  "\\\\$values": {
					  "\\\\$values": {
						  "\\\\$values": {
							  "\\\\$values": {
								  "\\\\$values": {
									  "\\\\$values": {
										  "$id": "35",
										  "$values": [],
										  "_type": "Object[]"
									  }
								  }
							  }
						  }
					  }
				  }
			  }
		  }
	  }
  },
  "_type": "ObjectMap"
}
```

For example same variable in version 3.1.3 looks like
```json
"PersistentVariablesDictionary": {
		"edc8e2ef3fb44a24bf829b1b15468b1b": {
			"$id": "72",
			"$values": [],
			"_type": "System.Collections.Generic.List`1[[System.String, System.Private.CoreLib]], System.Private.CoreLib"
		},
		"_type": "System.Collections.Generic.Dictionary`2[[System.String, System.Private.CoreLib],[System.Object, System.Private.CoreLib]], System.Private.CoreLib"
	}
```
Variable declaration:
`await context.ScheduleActivityAsync(
    Body, 
    ResumeAsync, 
    null, 
    new Variable[] 
    { 
        new(VariableConstants.Outcomes, new List<string>()) { StorageDriverType = typeof(WorkflowStorageDriver) }
    } );`

**Comment by sfmskywalker** [comment]
Hi @Nordran , would you be able to give it a try with the latest 3.3 preview and using the new `WorkflowInstanceStorageDriver` instead of the `WorkflowStorageDriver` (which will be deprecated)?

**Comment by sfmskywalker** [comment]
I have just tested this with the latest code in the main branch (built into 3.3.0-preview packages), where I'm able to work with array variables that are serialized and deserialized correctly. Here's the sample workflow I tested with:

[list-variables.json](https://github.com/user-attachments/files/16911725/list-variables.json)

I'll close this issue as solved, but feel free to continue the conversation. Thanks for your participation!
